Bio/Entrez/Parser.py

```diff
@@ -337,9 +337,9 @@ def read(self, handle):
             from io import StringIO
 
             if isinstance(handle, StringIO):
-                from Bio._py3k import _as_bytes
+                from Bio._utils import as_bytes
 
-                handle = BytesIO(_as_bytes(handle.read()))
+                handle = BytesIO(as_bytes(handle.read()))
```

Comment:
I am not sure why we need this; users should pass a BytesIO object, not a StringIO object to the `read` function. I can't think of a scenario in which the user /has/ to pass a StringIO object.

Reply:
A common use case would be user saved an XML file to disk, and just did ``open(xml_filename)`` which would be a unicode handle now.

Reply:
Then `Entrez.read` should raise an Exception. We should not be second-guessing the user.

Reply:
If it fails with a clear message, that's fine by me. I would lean to doing this in a follow up pull request so as not to delay this one editing many files.

Reply:
See this test in `test_Entrez_parser.py`:
```
    def test_text_handle(self):
        """Test parsing a handle opened in text mode."""
        with open("Entrez/einfo1.xml", "rt") as handle:
            record = Entrez.read(handle)
        self.assertIn("DbList", record)
```
or this one:
```
    def test_StringIO(self):
        """Test parsing a StringIO handle (unicode not bytes)."""
        with open("Entrez/einfo1.xml", "rt") as in_handle:
            data = in_handle.read()
        handle = StringIO(data)
        record = Entrez.read(handle)
        self.assertIn("DbList", record)
        handle.close()
```
and the documentation has this example:
```
>>> from Bio import Entrez
>>> Entrez.email = "A.N.Other@example.com"  # Always tell NCBI who you are
>>> handle = open("Homo_sapiens.xml")
>>> records = Entrez.parse(handle)
```

Reply:
All those examples open the file in text mode, giving unicode string handles - not bytes handles.

Reply:
Exactly. These should all be opening the file in binary mode. The call to `_as_bytes` obscures this.

Reply:
Could we leave this as is for now, merge this pull request, and then let you tackle this? It seems a bit much to ask @Andrey-Raspopov to do it here.

Reply:
I've removed version >= 3, since it'll be >= 3 anyway. Also I've changed initial commentary to TODO, since it definitely looks like a work for another PR, but also it looks like a work that should be done. I can do that after we are finished with py3k or I can leave to it to @mdehoon.

Reply:
I can handle Bio.Entrez but please discard all changes to Bio.Entrez (and any other module still under discussion) from this pull request.